source/extensions/filters/http/compressor/compressor_filter.cc

```diff
@@ -179,6 +179,8 @@ CompressorPerRouteFilterConfig::CompressorPerRouteFilterConfig(
       // Consequently, if `response_direction_config.common_direction_config.enabled` ever gets
       // added, its absence must enable compression.
       response_compression_enabled_ = true;
+      remove_accept_encoding_header_ =
+          config.overrides().response_direction_config().remove_accept_encoding_header();
```

Comment:
This will need to be guarded by `if(config.overrides().response_direction_config().has_remove_accept_encoding_header())`.